lib/streamlit/elements/lib/dialog.py

```diff
@@ -96,9 +96,6 @@ def _create(
 
         dialog._delta_path = delta_path
         dialog._current_proto = block_proto
-        # We add a sleep here to give the web app time to react to the update. Otherwise,
-        #  we might run into issues where the dialog cannot be opened again after closing
-        time.sleep(0.05)
```

Comment:
@kmcgrady @lukasmasuch I believe the solution with an `update_id` field enables us to get rid of the `time.sleep` which we wanted to get rid of anyways

Reply:
yay no sleeps!